### DISCUSSION TITLE
Construindo uma imagem docker após alterações

### DISCUSSION THREAD

**Discussion by JefPresotto** [Stirling-Tools/Stirling-PDF#317] [Q&A]
Utilizo a versão java, mas estou querendo migrar para a versão docker. Ja fiz algumas alterações nas paginas, como cores, nomes, etc, e gostaria de transferir isso para um conteiner docker. Como posso "compilar" a versão que ja utilizo?
Ou preciso refazer as alterações em uma imagem limpa? Caso seja este o caso, alguem pode me explicar como proceder?

Obrigado!

**Comment by Frooodle** [comment]
So if you made own changes you could build your own docker image (I can give you commands if you want) 
Or you can run my docker image via the compose/run script but overwrite the jar file with yours via docker volume either work work fine 

Also just asking... are any of the changes you have done beneficial to others do you think? If they are you should try share them and raise a PR!
Or share the ideas and we can talk about it such as customisable text or things

**Comment by JefPresotto** [comment]
Olá, obrigado pela resposta rápida.
Sim, acredito que pode ser interessante para todos, pois também comentei os códigos (para entender cada modulo) e isso pode facilitar o tratamento de itens básicos. Também adicionei anotações para os usuários em cada card, para facilitar o uso das ferramentas. Também estou finalizando um arquivo Help para os usuários, ou seja, um pdf que o usuario pode fazer o download e aprender a usar cada uma das ferramentas, que também vou disponibilizar. Também adicionei um compasso (régua) no card Adicionar Marca d'agua para orientar usuários básicos na questão de ângulo, apesar de simples, muitos não entendem. 

Gostaria de converter em um container docker pois é muito mais pratico e facil, e subir para o repositório. Caso possa me ajudar nisso, vou ficar muito grato, pois estou estudando bem o docker, porém algumas coisas ainda preciso de ajuda.

Este projeto realmente é fantástico e é uma iniciativa muito boa. Gostaria de ajudar com o que puder. Apesar das modificações serem modestas, todas são baseadas em necessidades de usuários que estão utilizando de forma massiva, e acredito que isso pode servir em todos os lugares.

![Screenshot_1](https://github.com/Frooodle/Stirling-PDF/assets/48888772/ecca79de-5c04-4f53-abf7-6db9574af5af)
![Screenshot_2](https://github.com/Frooodle/Stirling-PDF/assets/48888772/0fc9b2c2-9d78-445e-a58a-5df18d66aa64)
![Screenshot_3](https://github.com/Frooodle/Stirling-PDF/assets/48888772/6d5dc580-3e8f-4578-a59f-101b8d6001b4)
![Screenshot_4](https://github.com/Frooodle/Stirling-PDF/assets/48888772/d56cb7fc-b894-4497-84a1-4065c05010ff)
![Screenshot_6](https://github.com/Frooodle/Stirling-PDF/assets/48888772/d6ff50d1-956e-46a0-a6fe-e98a3b254796)

**Reply by Frooodle** [reply]
Interesting ideas!
I like the idea of adding custom footer links people can add and reference such as helpdesk etc for one.
same with custom about screens
I think the help sections for pages is a must
I am working on adding documentation to https://stirlingpdf.com/ at the moment but its a Work in progress.... Documentation is always hard :)

Watermark and other angle related things I never thought about addings tips so thats good feedback :) 

I would like to add that the copyright statement for this website is very wrong though and i hope you have some mention of its stirling-pdf github roots

**Reply by Frooodle** [reply]
As for docker help, im happy to help, are you on windows or linux/mac?

**Comment by JefPresotto** [comment]
Eu utilizo o ubuntu 22.04. 
Aqui temos muitas aplicações que rodam no datacenter e que utilizam aquele texto ali descrito. Mas como essa aplicação não é nossa, não será utilizando aquele texto, ele apenas esta como texto ocupando o espaço, porque ele só esta ali porque é uma cópia padrão de nossos serviços web com exceção da personalização do nome. Como vou subir o docker no repositório publico, voce pode conferir ele com todas as alterações na imagem e ajudar no texto "About". Vou citar todas as fontes do projeto, pois é um grande projeto.

**Comment by JefPresotto** [comment]
Bom dia amigo,
Aguardo instruções de apoio logo que tiver oportunidade.
Obrigado novamente!

**Reply by Frooodle** [reply]
#1  ``` chmod +x ./gradlew  ```
#2  ``` ./gradlew clean build ```
#3  install docker and setup username and repo 
#4  ``` docker build -t {docker username}/{docker repo}:{tagName} -f ./Dockerfile . ```
like ``` docker build -t froodle/stirling-pdf:0.0.1 -f ./Dockerfile . ``` 

#5 run new docker build
change  {docker username}/{docker repo}:{tagName} to your docker stuff
```

docker run -d \
  -p 8080:8080 \
  -v ./Striling-PDF/OCR:/usr/share/tesseract-ocr/4.00/tessdata \
  -v ./Striling-PDF/configs:/configs \
  --name stirling-pdf \
  {docker username}/{docker repo}:{tagName}
  
```

**Reply by Frooodle** [reply]
Hopefully that works let me know if you have any problems or need more details, this only is locally where you build the docker image locally, if you want to push it and run it on a different machine you need to do a push too which i can help with if needed

**Comment by JefPresotto** [comment]
Muito obrigado,
Fiz os testes e funcionou, apenas substitui a copia das fontes .otf por .woff2 no Dockerfile, pois estava travando a geração da imagem.
Tenho uma duvida com relação a estes dois parametros abaixo: 
  -v ./Striling-PDF/OCR:/usr/share/tesseract-ocr/4.00/tessdata \
  -v ./Striling-PDF/configs:/configs \
Qual a função deles? e existe alguma maneira de apontar definitivamente o caminho do tesseract?
Pois na pasta do tesseract ja constam as linguagens adicionais (.traineddata), porém ao rodar o docker, todas as modificação são carregadas com sucesso, menos as linguagens, a única que aparece é a english.

E gostaria do auxilio para fazer o push para o repositorio. 

Muito obrigado pelo seu tempo.
